brainiak/fcma/classifier.py

```diff
@@ -133,16 +133,121 @@ def fit(self, X, y):
+                and self.clf.kernel == 'precomputed':
+            assert self.training_data_ is not None, \
+                'when using precomputed kernel of SVM, ' \
+                'all training data must be provided'
+            num_training_samples = self.training_data_.shape[0]
+            data = np.zeros((num_test_samples, num_training_samples),
+                            np.float32,
+                            order='C')
+            corr_data = corr_data.reshape(num_test_samples,
+                                          num_voxels * num_voxels)
+            # compute the similarity matrix using corr_data and training_data
+            blas.compute_single_matrix_multiplication('T', 'N',
+                                                      num_training_samples,
+                                                      num_test_samples,
+                                                      num_voxels * num_voxels,
+                                                      1.0,
+                                                      self.training_data_,
+                                                      num_voxels * num_voxels,
+                                                      corr_data,
+                                                      num_voxels * num_voxels,
+                                                      0.0,
+                                                      data,
+                                                      num_training_samples)
+            logger.debug(
+                'similarity matrix computation done'
+            )
+        else:
+            data = corr_data.reshape(num_test_samples,
+                                     num_voxels * num_voxels)
+        return data
+
+    def fit(self, X, y):
+        """ use correlation data to train a model
+
+        first compute the correlation of the input data,
+        and then normalize within subject
+        if more than one sample in one subject,
+        and then fit to a model defined by self.clf.
+
+        Parameters
+        ----------
+        X: a list of numpy array in shape [num_TRs, num_voxels]
+            X contains the activity data filtered by top voxels
+            and prepared for correlation computation.
+            assuming all elements of X has the same num_voxels value
+        y: labels, len(X) equals len(Y)
+
+        Returns
+        -------
+        self: return the object itself
+        """
+        time1 = time.time()
+        assert len(X) == len(y), \
+            'the number of samples does not match the number labels'
+        num_samples = len(X)
+        num_voxels = X[0].shape[1]  # see assumption above
+        self.num_voxels_ = num_voxels
+        self.num_samples_ = num_samples
+        # correlation computation
+        corr_data = self._prepare_corerelation_data(X)
+        # normalization
+        corr_data = self._normalize_correlation_data(corr_data,
```

Comment:
You should avoid reusing names. How about something like `normalized_corr_data`?